sitemap/adder_test.go

```diff
@@ -0,0 +1,101 @@
+package sitemap_test
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/ONSdigital/dp-sitemap/sitemap"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestAdder(t *testing.T) {
+	Convey("When xml decode returns an error", t, func() {
+		oldSitemap := strings.NewReader("<<<")
+
+		a := &sitemap.DefaultAdder{}
+		filename, err := a.Add(oldSitemap, sitemap.URL{})
+
+		Convey("Adder should return correct error", func() {
+			So(err.Error(), ShouldContainSubstring, "failed to decode old sitemap")
+			So(err.Error(), ShouldContainSubstring, "XML syntax error")
+		})
+		Convey("Temporary sitemap file should be created and then cleaned up", func() {
+			So(filename, ShouldContainSubstring, "sitemap")
+			_, err := os.Stat(filename)
+			So(err.Error(), ShouldContainSubstring, "no such file or directory")
+		})
+	})
+	Convey("When old sitemap is empty", t, func() {
+		oldSitemap := strings.NewReader("")
+
+		a := &sitemap.DefaultAdder{}
+		filename, err := a.Add(oldSitemap, sitemap.URL{Loc: "a", Lastmod: "b"})
+		defer os.Remove(filename)
+
+		Convey("Adder should return with no error", func() {
+			So(err, ShouldBeNil)
+		})
+		Convey("Temporary sitemap file should be created and available", func() {
+			So(filename, ShouldContainSubstring, "sitemap")
+			_, err := os.Stat(filename)
+			So(err, ShouldBeNil)
+		})
+		Convey("Sitemap should be valid and include the received url", func() {
+			sitemap, err := os.ReadFile(filename)
```

Comment:
The variable named sitemap has the same name as the package, which could cause problems down the line.
```suggestion
			sitemapFile, err := os.ReadFile(filename)
```